server/channels/api4/channel.go

```diff
@@ -1241,15 +1241,21 @@ func searchArchivedChannelsForTeam(c *Context, w http.ResponseWriter, r *http.Re
 	}
 
 	var channels model.ChannelList
+	var member *model.TeamMember
 	var appErr *model.AppError
 	if c.App.SessionHasPermissionToTeam(*c.AppContext.Session(), c.Params.TeamId, model.PermissionListTeamChannels) {
 		channels, appErr = c.App.SearchArchivedChannels(c.AppContext, c.Params.TeamId, props.Term, c.AppContext.Session().UserId)
 	} else {
 		// If the user is not a team member, return a 404
-		if _, appErr = c.App.GetTeamMember(c.AppContext, c.Params.TeamId, c.AppContext.Session().UserId); appErr != nil {
+		member, appErr = c.App.GetTeamMember(c.AppContext, c.Params.TeamId, c.AppContext.Session().UserId)
+		if appErr != nil {
 			c.Err = appErr
 			return
 		}
```

Comment:
I don't understand the permission check here. If the user doesn't have the permission `PermissionListTeamChannels`, why do we check the membership? Should all members of a team have the `PermissionListTeamChannels` permission?

Reply:
Thanks for flagging this @hanzei; indeed this appears to be an improperly applied remnant pattern from `searchChannelsForTeam`, which does have a valid use for the non-permission fallback search.

I've removed the else path